### PR TITLE
fix(gcd): handle edge cases around zero 

### PR DESCRIPTION
**Pull request by goldenMetteyya**
Fix edge case for negative zero

### CODE REVIEW

**Comment by dignifiedquire** [comment]
Interesting, this looks right to me, could you please run `rustfmt` on the file. Do you have a test case that reproduces this, would be good to have a test for this

**Comment by goldenMetteyya** [comment]
Hey @dignifiedquire  Sorry for the late reply, I got locked out of GitHub because of the 2fa and I couldn't get the text abroad, lesson learned:) I will review this again. Could be something on how I am using the library I haven't figured that out.

thanks

**Comment by goldenMetteyya** [comment]
Hey @dignifiedquire , So I think the issue I face is when I try and use a negative number in an extended_gcd. When the BigInt unwraps into an BigUint it doesn't. I added a test case to show this. Maybe I am missing something.

thanks

**Comment by goldenMetteyya** [comment]
https://github.com/dignifiedquire/num-bigint/blob/master/src/bigint.rs#L2617

I see here that the to_biguint from a BigInt is none for negative values

**Comment by goldenMetteyya** [comment]
Hey @dignifiedquire ,

So I have found the issue overall. As the gcd and extended version is a port from golangs lib

https://github.com/golang/go/issues/28878

there is an issue about it and a patch

https://go-review.googlesource.com/c/go/+/164972/

but it wasn't merged as its breaking the api.

What do you think is best for this library, to make it native with this patch and update the code accordingly or have a separate function for it.

**Comment by goldenMetteyya** [comment]
the one test that fails passes in all except when 
` cargo test --verbose --no-default-features --features="std $feature"`

**Comment by goldenMetteyya** [comment]
Hey @dignifiedquire , 

I believe to have added the changes needed through a separate function. Based on how the api change might be, can be merged or needs to combined with the current interface.

thanks

**Comment by dignifiedquire** [comment]
@goldenMetteyya thank you for following up on this, I think we should definitely bring the patch in, ensuring all inputs work well. I'll review the api in a bit

**Comment by goldenMetteyya** [comment]
> @goldenMetteyya thank you for following up on this, I think we should definitely bring the patch in, ensuring all inputs work well. I'll review the api in a bit

Hey @dignifiedquire  any thoughts so I can close this PR?

**Comment by goldenMetteyya** [comment]
@goldenMetteyya would like to clean this up soon to make it more production ready.

File: src/lib.rs [hunk 1]
```diff
@@ -209,7 +210,7 @@ pub const VEC_SIZE: usize = 8;
 #[cfg(feature = "u64_digit")]
 pub const VEC_SIZE: usize = 4;
 
-mod big_digit {
+pub mod big_digit {
```
Comment (dignifiedquire) [inline src/lib.rs]: the internals of this should not be public

**Comment by goldenMetteyya** [comment]
Weired that I made the change and it still didn't recognize
